configs/common/packages.yaml

```diff
@@ -9,290 +9,255 @@
+      require: '@5.1.0'
     metis:
-      require: "+int64 +real64"
+      require: '+int64 +real64'
     mpich:
-      variants: ~hwloc +two_level_namespace
+      require: '~hwloc'
     nco:
-      version: ['5.1.6']
-      variants: ~doc
+      require: '@5.1.6 ~doc'
     # ncview - when adding information here, also check Orion
     # and Discover site configs
     nemsio:
-      version: ['2.5.4']
+      require: '@2.5.4'
     nemsiogfs:
-      version: ['2.5.3']
+      require: '@2.5.3'
     nccmp:
-      version: ['1.9.0.1']
+      require: '@1.9.0.1'
     ncio:
-      version: ['1.1.2']
+      require: '@1.1.2'
     netcdf-c:
       # If using 4.9.1, turn off byterange variant to fix compile error: ~byterange
-      require: "@4.9.2 +dap +mpi ~parallel-netcdf ~szip build_system=autotools"
+      require: '@4.9.2 +dap +mpi ~parallel-netcdf ~szip build_system=autotools'
     netcdf-cxx4:
-      version: ['4.3.1']
+      require: '@4.3.1'
     netcdf-fortran:
-      version: ['4.6.1']
+      require: '@4.6.1'
     # ninja - when adding information here, also check Discover site config
-    nlohmann-json:
-      version: ['3.10.5']
-    nlohmann-json-schema-validator:
-      version: ['2.1.0']
     odc:
-      version: ['1.4.6']
-      variants: ~fortran
+      require: '@1.4.6 ~fortran'
     openblas:
-      version: ['0.3.24']
-      variants: +noavx512
+      require: '@0.3.24 +noavx512'
     openmpi:
-      variants: ~internal-hwloc +two_level_namespace
-    # Pin openssl to avoid duplicate packages being built
+      require: '~internal-hwloc +two_level_namespace'
     openssl:
-      variants: +shared
+      require: '+shared'
     p4est:
-      require: "@2.8"
+      require: '@2.8'
     parallelio:
-      version: ['2.6.2']
-      variants: +pnetcdf
+      require: '@2.6.2 +pnetcdf'
     parallel-netcdf:
-      version: ['1.12.3']
+      require: '@1.12.3'
     pflogger:
+      require: '+mpi'
       #version: ['1.12.0']
-      variants: +mpi
     pixman:
-      variants: +pic
-    # Do not build pkgconf - https://github.com/jcsda/spack-stack/issues/123
-    pkgconf:
-      buildable: False
+      require: '+pic'
     prod-util:
-      version: ['2.1.1']
+      require: '@2.1.1'
     proj:
-      version: ['8.1.0']
-      variants: ~tiff
+      require: '~tiff'
     python:
-      require: "@3.10.13"
+      require: '@3.10.13'
     py-attrs:
       # https://github.com/JCSDA/spack-stack/issues/740
-      version: ['21.4.0']
+      require: '@21.4.0'
     py-cartopy:
-      variants: +plotting
-      require: "@0.21.1"
+      require: '@0.21.1 +plotting'
     py-cryptography:
-      variants: +rust_bootstrap
+      require: '+rust_bootstrap'
     # Introduced in https://github.com/JCSDA/spack-stack/pull/894, pin py-cython
     # to avoid duplicate packages being built (cylc dependencies soft-want @3:)
     py-cython:
-      require: "@0.29.36"
+      require: '@0.29.36'
+    # To avoid duplicate packages
+    py-flit-core:
```

Comment:
I need to test this. Chances are that this will break the neptune-dev +python environment, which currently isn't tested in CI

Reply:
this seems to be ok